File: src/main/java/com/leijendary/spring/microservicetemplate/config/properties/InfoProperties.java
package com.leijendary.spring.microservicetemplate.config.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import springfox.documentation.service.Contact;
import springfox.documentation.service.VendorExtension;

import java.util.ArrayList;
import java.util.List;

@ConfigurationProperties(prefix = "info")
@Data
public class InfoProperties {

    private App app;
    private Api api;

    @Data
    public static class App {

        private String organization;
        private String groupId;
        private String artifactId;
        private String name;
        private String description;
        private String version;
    }

    @Data
    public static class Api {

        private String termsOfServiceUrl;
        private Contact contact;
        private String license;
        private String licenseUrl;

        @SuppressWarnings("rawtypes")
        private List<VendorExtension> vendorExtensions = new ArrayList<>();
    }
}

File: src/main/kotlin/com/leijendary/spring/template/core/config/properties/InfoProperties.kt
package com.leijendary.spring.template.core.config.properties

import io.swagger.v3.oas.models.info.Contact
import io.swagger.v3.oas.models.info.License
import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties("info")
class InfoProperties {
    var app = App()
    var api = Api()

    class App {
        var organization: String = ""
        var group: String = ""
        var name: String = ""
        var description: String = ""
        var version: String = ""
    }

    class Api {
        var termsOfService: String = ""
        var contact: Contact? = null
        var license: License? = null
        var extensions: Map<String, Any> = HashMap()
    }
}
